src/librbd/api/Image.cc

```diff
@@ -499,6 +513,11 @@ int Image<I>::list_descendants(
     image.trash = it->second.second;
   }
 
+  // remove stale child image which its pool does not exist
```

Comment:
Nit: instead of removing them, I would just leave the `pool_name` and `image_name` fields blank and then modify the rbd CLI to output `<missing>` or something similar if the field if blank.